scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/dragen/BCLConvert.java

```diff
@@ -0,0 +1,266 @@
+          // look like?
+          String[] indices = demuxLine[2].split("-");
+          DragenAnalysisUnit dragenAnalysisUnit;
+          if (indices.length == 2) {
+            dragenAnalysisUnit =
+                bclConvertWorkflowRun.get(demuxLine[1], demuxLine[0], indices[0], indices[1]);
+          } else if (indices.length == 1) {
+            dragenAnalysisUnit =
+                bclConvertWorkflowRun.get(demuxLine[1], demuxLine[0], indices[0], null);
+          } else {
+            throw new IllegalStateException(
+                "Demux indices length for " + demuxLine[1] + " is " + indices.length);
+          }
+          if (dragenAnalysisUnit == null) dragenAnalysisUnit = new DragenAnalysisUnit();
+          int readCount = Integer.parseInt(demuxLine[3]);
+          // Set the same read count for all files in analysis
+          dragenAnalysisUnit
+              .getFiles()
+              .forEach(file -> ((FastqAnalysisFile) file).setReadCount(readCount));
+          bclConvertWorkflowRun.put(dragenAnalysisUnit);
+        }
+      } else {
+        log.info("No Demultiplex_Stats.csv for {}", rootDir);
+      }
+    } else {
+      log.info("No fastq_list.csv for {}", rootDir);
+    }
+
+    // Check against samplesheet for okayness
+    boolean isOk = true;
+    for (SamplesheetBCLConvertDataEntry item :
+        ((SamplesheetBCLConvertSection) samplesheet.getByName("BCLConvert")).getData()) {
+      DragenAnalysisUnit dragenAnalysisUnitItem =
+          bclConvertWorkflowRun.get(
+              item.getSampleId(),
+              item.getLane(),
+              item.getIndex(),
+              reverseComplement(item.getIndex2()));
+      if (dragenAnalysisUnitItem == null || dragenAnalysisUnitItem.isEmpty()) {
+        isOk = false;
+        break;
+      } else {
+        for (AnalysisFile itemFile : dragenAnalysisUnitItem.getFiles()) {
+          // Unfortunately there is no better source for OverrideCycles
+          ((FastqAnalysisFile) itemFile).setOverrideCycles(item.getOverrideCycles());
+          if (itemFile.getPath() == null
+              || itemFile.getPath().toString().isBlank()
+              || itemFile.getCrc32Checksum() == null
+              || itemFile.getCrc32Checksum().isBlank()
+              || ((FastqAnalysisFile) itemFile).getOverrideCycles() == null
+              || ((FastqAnalysisFile) itemFile).getReadCount() == null
+              || ((FastqAnalysisFile) itemFile).getReadNumber() == null) {
+            isOk = false;
+            break;
+          }
+        }
+      }
+    }
+    if (isOk) bclConvertWorkflowRun.complete();
+    return bclConvertWorkflowRun;
+  }
+
+  private static AnalysisFile fastqFromFilename(File rootDir, String fileName, int readNumber)
+      throws IOException {
+    Path fullPath = Paths.get(rootDir.getPath(), "/Data/BCLConvert/fastq/", fileName);
+    if (!Files.exists(fullPath)) {
```

Comment:
An empty string (and probably null) for `fileName` would result in checking that the `fastq` directory exists. Should probably check for null and empty string at the start and return null